### ISSUE TITLE
HSEP: Improve Plotting Quality of Life

### ISSUE THREAD

**Issue by CSSFrancis** [hyperspy/hyperspy#3327]
#### Describe the functionality you would like to see.
I wanted to make an issue that we could use to track the progress and maybe help direct efforts related to improving the plotting quality in hyperspy.  I think that currently things work, but things could be much more effective/ faster than things currently.  There is also lots of interest in adjusting/customizing navigator and signal plots much more than is currently possible. 

- [ ] #3325 
- [ ] #3326 
- [ ] Overlaying signals (plotting two signals/ two navigators) on top of each other with some opacity.
- [ ] Returning signal and navigator figures when calling plot. (Allows for saving/adjusting etc.)
- [ ] Figure composition: #3343 
   - [ ] Allow passing predefined axes
   - [ ] Improve figure size based on aspect ratio of data

#### Additional information
Feel free to edit this post as we get a better idea of what we want to do.

@ericpre do you want to add things you might think would be good to handle?

**Comment by CSSFrancis** [comment]
It might also be good to look into https://matplotlib.org/stable/gallery/subplots_axes_and_figures/subfigures.html#figure-subfigures and think about how we might handle something like having multiple navigators in 2D etc.  That seems like a useful thing to have as an option.

**Comment by ericpre** [comment]
Indeed, being able to specify `axes` would indeed be very nice as this would allow to compose figure. This would also to plot navigation and signal on same figure, which could be used for decomposition visualisation, etc. (see for https://hyperspy.org/hyperspy-doc/current/user_guide/visualisation.html#id28)
